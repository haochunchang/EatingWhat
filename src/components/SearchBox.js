import React from "react";

const SearchBox = ({ searchChange }) => {
  return (
    <div className="tc pa2">
      <label>
        Have something in mind?
        <input
          className="pa3 ba bg-washed-red"
          type="search"
          placeholder="search foods"
          onChange={searchChange}
        />
      </label>
    </div>
  );
};

export default SearchBox;
