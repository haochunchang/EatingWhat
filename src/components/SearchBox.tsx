import React from "react";

interface SearchBoxProps {
  searchChange(event: React.SyntheticEvent<HTMLInputElement>): void;
}

const SearchBox = ({ searchChange }: SearchBoxProps) => {
  return (
    <div className="tc pa2">
      <label>
        Have something in mind?
        <input
          className="pa3 ba bg-washed-red"
          type="search"
          placeholder="search foods"
          onChange={searchChange}
        />
      </label>
    </div>
  );
};

export default SearchBox;
