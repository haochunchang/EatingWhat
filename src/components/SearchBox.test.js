import React from "react";
import { shallow } from "enzyme";
import SearchBox from "./SearchBox";

describe("SearchBox", () => {
  it("matches SearchBox snapshot", () => {
    const searchText = "testing";
    expect(shallow(<SearchBox searchChange={searchText} />)).toMatchSnapshot();
  });
});
