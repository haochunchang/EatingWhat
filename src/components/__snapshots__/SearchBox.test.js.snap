// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox matches SearchBox snapshot 1`] = `ShallowWrapper {}`;
